tests/mypy_test.py

```diff
@@ -280,6 +294,94 @@ def is_probably_stubs_folder(distribution: str, distribution_path: Path) -> bool
     return distribution != ".mypy_cache" and distribution_path.is_dir()
 
 
+class TestResults(NamedTuple):
+    exit_code: int
+    files_checked: int
+
+
+def test_typeshed(code: int, major: int, minor: int, args: argparse.Namespace) -> TestResults:
```

Comment:
This function is rather long, what about splitting it into separate functions for the stdlib, third_party, tests/, and test_cases/?

Reply:
Sure!